File: Updater/converters/DerivedGeneralCategory.rb
=begin
 
 DerivedGeneralCategory.rb
   ©︎ 2018 YOCKOW.
     Licensed under MIT License.
     See "LICENSE.txt" for more information.
 
=end


module DerivedGeneralCategory
  PATH = 'Sources/UnicodeSupplement/__data__/UCD/DerivedGeneralCategory.swift'
  URLs = [
    URI.parse('https://www.unicode.org/Public/UCD/latest/ucd/extracted/DerivedGeneralCategory.txt')
  ]
end

module DerivedGeneralCategory; class << self
  def write(remote_io, file)
    table = Table.new(remote_io)
    failed("No data about DerivedCombiningClass.") if table.count < 1
    
    gc_table = UnicodeTable.new(table, 1)
    file.write(gc_table.to_swift_assoc_array('gc', 'String'))
  end
end; end

File: utils/SwiftUnicodeSupplementUpdater/Sources/UnicodeSupplementUpdater/DerivedGeneralCategory.swift
/* *************************************************************************************************
 DerivedGeneralCategory.swift
   © 2020 YOCKOW.
     Licensed under MIT License.
     See "LICENSE.txt" for more information.
 ************************************************************************************************ */
 
import Foundation
import UnicodeSupplement

open class DerivedGeneralCategory: UCDPropertiesCodeUpdaterDelegate<Unicode.GeneralCategory> {
  open override var prefix: String { return "gc" }
  
  open override var sourceURLs: Array<URL> {
    return [
      URL(string: "https://www.unicode.org/Public/UCD/latest/ucd/extracted/DerivedGeneralCategory.txt")!
    ]
  }
  
  open override func reduce(columns: [String]) throws -> Unicode.GeneralCategory {
     return .init(abbreviated: columns.first!)
   }
   
   open override func describe(value: Unicode.GeneralCategory) -> String {
     return ".\(String(describing: value))"
   }
}
